src/CalcManager/CEngine/scicomm.cpp

```diff
@@ -714,6 +733,47 @@ void CCalcEngine::ProcessCommandWorker(OpCode wParam)
         HandleErrorCommand(wParam);
         break;
 
+    case IDC_RAND:
+        if (!m_fIntegerMode)
+        {
+            CheckAndAddLastBinOpToHistory(); // rand is like entering the number
+
+            random_device rd;
+            mt19937 gen(rd());
```

Comment:
NIT: We should create only 1 instance of `mt19937 gen` (first time we use it) instead of creating a new generator on each call to ensure that the distribution of results is done as expected.

Reply:
Moved the initialization to CalcEngine.h. I couldn't figure out how to check if it had already been initialized since I wanted to avoid initializing it on start up.

I tried this code in CalcEngine.h but kept getting the same rand number repeatedly. Any ideas?

```
    bool m_isRandomGeneratorSeeded;
    std::mt19937 m_randomGeneratorEngine;

    std::mt19937 GetRandomGeneratorEngine()
    {
        if (!m_isRandomGeneratorSeeded)
        {
            std::random_device rd;
            m_randomGeneratorEngine = std::mt19937(rd());
            m_isRandomGeneratorSeeded = true;
        }
        return m_randomGeneratorEngine;
    }
```

Reply:
I don't have this issue, here is my code:

**RandomGenerator.h**
```h
#pragma once
#include <random>

class RandomGenerator
{
public:
	double GenerateNumber();
private:
	std::unique_ptr<std::mt19937> m_randomGeneratorEngine;
	std::unique_ptr<std::uniform_real_distribution<>> m_distr;
};
```

**RandomGenerator.cpp**
```cpp
#include "pch.h"
#include "RandomGenerator.h"

using namespace std;

double RandomGenerator::GenerateNumber()
{
	if (m_randomGeneratorEngine == nullptr)
	{
		random_device rd;
		m_randomGeneratorEngine = std::make_unique<std::mt19937>(rd());
		m_distr = std::make_unique<std::uniform_real_distribution<>>(0, 1);
	}
	return (*m_distr.get())(*m_randomGeneratorEngine.get());
}
```



Sample app here: https://1drv.ms/u/s!AkrL-mAvq0QFs8obJ3vwbChR5LvFZA?e=dmBLED

Reply:
Thanks, this works!